### ISSUE TITLE
Self-referencing submodule?

### ISSUE THREAD

**Issue by cniggeler** [scivision/fortran-submodule#2]
Hi, not an issue per se, but a question about the code: in basic_sub.f90, what does this code do?
`submodule (mother) mother`
`end submodule mother`

I note there is a module named "mother" in basic.f90.

The code compiles and produces the same output if you comment out the submodule mother thing.

**Comment by scivision** [comment]
Thanks, I made this its own example in src/same.f90

**Comment by cniggeler** [comment]
In regards to your comment in same.f90: "Is this in Fortran standard?" - it is not, at least in the draft standard I use. The section on submodules in the F08 standard is only 21 lines long!

However, document ISO/IEC TR 19767:2004(E), "Information technology — Programming languages — Fortran — Enhanced Module Facilities", states: "Submodule identifiers are global identifiers, but since they consist of a module name and a descendant submodule name, the name of a submodule can be the same as the name of another submodule so long as they do not have the same ancestor module."

It appears gfortran, at least, extends this meaning to modules as well. To test this, I modifiled same.f90 to include another submodule, and this compiles and works as expected.

I tried to name "other" as "same" yet again, and gfortran produced this error message: 
Error: Global name 'same.same' at (1) is already being used as a MODULE at (2)
f951.exe: Fatal Error: Can't USE the same submodule we're building!
compilation terminated.

which seems in agreement with the ISO doc, since they would have the same ancestor module.

```
module same
   interface
      module subroutine hello()
      end subroutine hello
      module subroutine hello2()
      end subroutine hello2
   end interface
end module same

submodule (same) same
contains
   module procedure hello
      print *, "submodule same has parent module same"
   end procedure hello
end submodule same

submodule (same:same) other
contains
   module procedure hello2
      print *, "sub-submodule other has ancestor same and parent same!"
   end procedure hello2
end submodule other

program same_name
   use same
   implicit none
   call hello()
   call hello2()
end program


```
